packages/remix-react/browser.tsx

```diff
@@ -27,7 +30,7 @@ export interface RemixBrowserProps {}
 export function RemixBrowser(_props: RemixBrowserProps): ReactElement {
   let historyRef = React.useRef<BrowserHistory>();
   if (historyRef.current == null) {
-    historyRef.current = createBrowserHistory({ window });
+    historyRef.current = createBrowserHistory({ window, v5Compat: true });
```

Comment:
The new histories in `@remix-run/router` need to be put into a `v5Compat` mode in order to trigger listeners on push/replace